daras_ai_v2/base.py

```diff
@@ -841,42 +876,33 @@ def _render_publish_form(
             )
         raise gui.RedirectException(pr.get_app_url())
 
-    def _render_workspace_selector(self, *, key: str) -> "Workspace":
-        if not self.can_user_see_workspaces():
-            return self.current_workspace
-
+    def _get_workspace_options(self) -> dict[int, Workspace]:
         workspace_options = {w.id: w for w in self.request.user.cached_workspaces}
-
         if self.current_pr.workspace_id and self.can_user_edit_published_run(
             self.current_pr
         ):
             # make current_pr.workspace the first option
             workspace_options = {
                 self.current_pr.workspace_id: self.current_pr.workspace
             } | workspace_options
+        return workspace_options
```

Comment:
this was converted into a function of its own to reuse the logic in:
- to determine whether the selector should be shown
- inside the selector